src/Processors/QueryPlan/Optimizations/QueryPlanOptimizationSettings.h

```diff
@@ -12,77 +12,52 @@ struct Settings;
 struct QueryPlanOptimizationSettings
 {
     /// Allows to globally disable all plan-level optimizations.
-    /// Note: Even if '= true', individual optimizations may still be disabled via below settings.
+    /// Note: Even if set to 'true', individual optimizations may still be disabled via below settings.
     bool optimize_plan = true;
 
     /// If not zero, throw if too many optimizations were applied to query plan.
     /// It helps to avoid infinite optimization loop.
     size_t max_optimizations_to_apply = 0;
 
-    /// If moving-up-of-array-join optimization is enabled.
-    bool lift_up_array_join = true;
+    /// ------------------------------------------------------
+    /// Enable/disable plan-level optimizations
 
-    /// If moving-limit-down optimization is enabled.
-    bool push_down_limit = true;
+    /// --- Zero-pass optimizations (Processors/QueryPlan/QueryPlan.cpp)
+    bool remove_redundant_sorting = true;
 
-    /// If split-filter optimization is enabled.
+    /// --- First-pass optimizations
+    bool lift_up_array_join = true;
+    bool push_down_limit = true;
     bool split_filter = true;
-
-    /// If merge-expressions optimization is enabled.
     bool merge_expressions = true;
-
-    /// If merge-filters optimization is enabled.
     bool merge_filters = true;
-
-    /// If filter push down optimization is enabled.
     bool filter_push_down = true;
```

Comment:
I agree that the prior documentation was not informative.

Reply:
In particular, all these settings are documented exhaustively in Core/Settings.cpp so I was removing some redundancy.